akamai/cache-client.go

```diff
@@ -52,25 +52,28 @@ type purgeResponse struct {
 	PurgeID          string `json:"purgeId"`
 }
 
-// CachePurgeClient talks to the Akamai CCU REST API. It is safe to make
+// CachePurgeClient is an interface for testing purposes.
+type CachePurgeClient interface {
+	Purge(queueEntries [][]string) error
+}
```

Comment:
Since the interface is only used in `cmd/akamai-purger` (it's a testing interface), it's more idiomatic to put the interface definition in `cmd/akamai-purger` and make it unexported. I know we don't follow this pattern perfectly in our codebase, but it's somewhat preferable.

The alternative would be to export the interface, and have `NewCachePurgeClient` return an instance of that interface, but leave `CachePurgeClientImpl` unexported. That's what you would choose if you were trying to hide the implementation, or leave it up to the `akamai` package to potentially return different implementations in different situations. That would be (to make up a term) a hiding interface.

One reason I consider this to be a testing interface: it doesn't include all the methods on `CachePurgeClient` (e.g. PurgeTags).